modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java

```diff
@@ -304,14 +304,19 @@ private static CompletableFuture<Integer> partitionUpdate(
 
         Assignment nextAssignment = nextAssignment(localPartitionStateMessageByNode, partAssignments);
 
+        boolean isNextEqualsToRestoredAsPlannedAssignments = partAssignments.size() == 1 && partAssignments.contains(nextAssignment);
+
         // There are nodes with data, and we set pending assignments to this set of nodes. It'll be the source of peers for
         // "resetPeers", and after that new assignments with restored replica factor wil be picked up from planned assignments
         // for the case of the manual update, that was triggered by a user.
         Iif invokeClosure = prepareMsInvokeClosure(
                 partId,
                 longToBytesKeepingOrder(revision),
                 Assignments.forced(Set.of(nextAssignment), assignmentsTimestamp).toBytes(),
-                Assignments.toBytes(partAssignments, assignmentsTimestamp)
+                // In case if nodes set is equal then we shouldn't schedule the same planned rebalance
```

Comment:
is equal to what?

Reply:
Old commentary, I guess the new one is more precise now